tslint-rules/buttonGroupOrderRule.ts

```diff
@@ -93,30 +82,50 @@ class ButtonGroupOrderWalker extends Lint.RuleWalker {
     }
 
     const buttonsWithTypeAttr = buttons.map(b => {
-      const typeAttr = b.attributes.properties.find(a =>
-        a.kind === ts.SyntaxKind.JsxAttribute && a.name.getText() === 'type'
+      const typeAttr = b.attributes.properties.find(
+        a =>
+          a.kind === ts.SyntaxKind.JsxAttribute && a.name.getText() === 'type'
       ) as ts.JsxAttribute | undefined
 
       let value = undefined
 
-      if (typeAttr && typeAttr.initializer && typeAttr.initializer.kind === ts.SyntaxKind.StringLiteral) {
+      if (
+        typeAttr &&
+        typeAttr.initializer &&
+        typeAttr.initializer.kind === ts.SyntaxKind.StringLiteral
+      ) {
         value = typeAttr.initializer.text
       }
 
-      return [ b, value ]
+      return [b, value]
     })
 
-    const primaryButtonIx = buttonsWithTypeAttr.findIndex(x => x[1] === 'submit')
+    const primaryButtonIx = buttonsWithTypeAttr.findIndex(
+      x => x[1] === 'submit'
+    )
 
     if (primaryButtonIx !== -1 && primaryButtonIx !== 0) {
       const start = node.getStart()
       const width = node.getWidth()
       const error = `Wrong button order in ButtonGroup.`
-      const explanation = 'ButtonGroups should have the primary button as its first child'
+      const explanation =
+        'ButtonGroups should have the primary button as its first child'
 
       const message = `${error} ${explanation}`
 
       this.addFailure(this.createFailure(start, width, message))
     }
   }
 }
+
+export class Rule extends Lint.Rules.AbstractRule {
```

Comment:
Is there a reason for re-ordering this rule last rather than first?

Reply:
I get this:

```
desktop/tslint-rules/buttonGroupOrderRule.ts
  33:13  error  'ButtonGroupOrderWalker' was used before it was defined  typescript/no-use-before-define
```

```ts
      return this.applyWithWalker(
        new ButtonGroupOrderWalker(sourceFile, this.getOptions())
//          ^^^^^^^^^^^^^^^^^^^^^^
      )
```

if I move it back.

Reply:
Ah, so `typescript/no-use-before-define` is a new rule. I'm not sure if `tslint` had an equivalent. This also affects existing application code - [seems like a reasonable thing](https://github.com/nzakas/eslint-plugin-typescript/blob/master/docs/rules/no-use-before-define.md) based on the docs.